asyncpg/connection.py

```diff
@@ -909,6 +930,25 @@ def _cancel_current_command(self, waiter):
 
         self._loop.create_task(cancel())
 
+    def _notice(self, message):
+        # `self._notice_cb` is checked internally and passed here as the `cb`
+
+        if self._proxy is None:
+            con_ref = self
+        else:
+            # See the comment in the `_notify` below
+            con_ref = self._proxy
+
+        for cb in self._notice_cb_set:
+            try:
+                cb(con_ref, message)
```

Comment:
And we'll update `notify` to use `call_soon` in a separate PR. That's something that we just oversaw.